apps/namadillo/src/slices/accounts/services.ts

```diff
@@ -0,0 +1,46 @@
+import { getIntegration } from "@namada/integrations";
+import { Account } from "@namada/types";
+import BigNumber from "bignumber.js";
+import { getSdkInstance } from "hooks";
+
+export const fetchAccounts = async (): Promise<readonly Account[]> => {
+  const namada = getIntegration("namada");
+  const result = await namada.accounts();
+  if (typeof result === "undefined") {
+    throw new Error("Integration error: accounts object is undefined");
+  }
+  return result;
+};
+
+export const fetchDefaultAccount = async (): Promise<Account | undefined> => {
```

Comment:
I think it will return `undefined` in the case we have the extension installed and no keys added. But I think we don't need to throw an error because of that, what do you think?

Reply:
maybe if instead of throwing an error on `fetchAccounts`, we just return an empty array, then could we only issue `fetchDefaultAccount` if `accounts` length > 0? Perhaps that would be a little more graceful in the event that the extension has zero accounts, but I think it could go either way. The user should def know that they should set up at least one account if they expect to do anything

Reply:
I've changed to return an empty array when querying `fetchAccounts`, however, I think it wouldn't be necessary to wait for its response before calling `fetchDefaultAccount`. I think returning `undefined` should be fine if the user has no default account set yet